src/benchmarks/micro/libraries/System.Text/Perf.StringBuilder.cs

```diff
@@ -178,6 +178,26 @@ public StringBuilder Append_ValueTypes()
             return builder;
         }
 
+        // on .NET 6+, interpolated string handlers make appending more efficient by avoiding boxing and using ISpanFormattable.
+        [Benchmark]
+        public StringBuilder Append_ValueTypes_Interpolated()
+        {
+            var builder = new StringBuilder();
+
+            var dateTime = new DateTime(2018, 12, 14);
+            var timeSpan = new TimeSpan(1, 2, 0);
+            var dateTimeOffset = new DateTimeOffset(dateTime, timeSpan);
+            var guid = new Guid(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);
+
+            // to amortize the cost of creating the value types we add them few times
+            builder.Append($"{guid}"); builder.Append($"{dateTime}"); builder.Append($"{timeSpan}"); builder.Append($"{dateTimeOffset}");
+            builder.Append($"{guid}"); builder.Append($"{dateTime}"); builder.Append($"{timeSpan}"); builder.Append($"{dateTimeOffset}");
+            builder.Append($"{guid}"); builder.Append($"{dateTime}"); builder.Append($"{timeSpan}"); builder.Append($"{dateTimeOffset}");
+            builder.Append($"{guid}"); builder.Append($"{dateTime}"); builder.Append($"{timeSpan}"); builder.Append($"{dateTimeOffset}");
```

Comment:
Is C# compiler smart enough to detect that the value types don't change their values and the each string needs to be interpolated just once? If so, this benchmark is measuring `builder.Append(string)` performance which is already covered here and we need to modify the values. 

```cs
var dateTime = new DateTime(2018, 12, 14);
var timeSpan = new TimeSpan(1, 2, 0);
var dateTimeOffset = new DateTimeOffset(dateTime, timeSpan);
var guid = new Guid(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11);

for (int i = 1; i <= 10; i++)
{
    builder.Append($"{guid} {dateTime} {timeSpan} {dateTimeOffset}");
    dateTime = date.AddSeconds(i);
}
```

Reply:
I don't think the compiler is _that_ smart to recognize such pattern, but I still changed the benchmark to use a loop and a single interpolated string with many holes (a more real-word case).